app/src/main/java/com/danosoftware/galaxyforce/services/googleplay/GooglePlayServices.java

```diff
@@ -0,0 +1,225 @@
+//            GoogleSignInAccount signedInAccount = result.getSignInAccount();
+////                GamesClient gamesClient = Games.getGamesClient(this, signedInAccount);
+////                gamesClient.setViewForPopups(this.findViewById(android.R.id.content));
+//        } else {
+//            String message = result.getStatus().getStatusMessage();
+//            if (message == null || message.isEmpty()) {
+//                message = "Unknown";
+//            }
+////                new AlertDialog.Builder(this).setMessage(message)
+////                        .setNeutralButton(android.R.string.ok, null).show();
+//        }
+
+
+
+
+//        if(completedTask.isSuccessful()) {
+//            GoogleSignInAccount signedInAccount = completedTask.getResult();
+//            Log.i(ACTIVITY_TAG, "signInResult:success");
+//            onConnected(signedInAccount);
+//        } else {
+//            String message = completedTask.getStatus().getStatusMessage();
+//            if (message == null || message.isEmpty()) {
+//                message = "Unknown";
+//            }
+//                new AlertDialog.Builder(mActivity).setMessage(message)
+//                        .setNeutralButton(android.R.string.ok, null).show();
+//        }
+//            Exception exception = completedTask.getException();
+//            if (exception instanceof ApiException) {
+//                final ApiException apiException = (ApiException) exception;
+//                final int exceptionCode = apiException.getStatusCode();
+//                final String exceptionCause = CommonStatusCodes.getStatusCodeString(exceptionCode);
+//                Log.w(ACTIVITY_TAG, String.format(
+//                        "signInResult:failure code={}. reason={}",
+//                        exceptionCode,
+//                        exceptionCause));
+//                connectedState = ConnectedState.DISCONNECTED;
+//            } else {
+//                Log.w(ACTIVITY_TAG, String.format(
+//                        "signInResult:failure reason={}",
+//                        exception.getMessage()));
+//            }
+//        }
+
+
+        try {
+            GoogleSignInAccount account = completedTask.getResult(ApiException.class);
+            Log.i(ACTIVITY_TAG, "signInResult:success");
+            onConnected(account);
+        } catch (ApiException e) {
+            Log.w(ACTIVITY_TAG, "signInResult:failed code=" + e.getStatusCode());
+            connectedState = ConnectedState.DISCONNECTED;
+//            signInSilently();
+        }
+    }
+
+    public void signOut() {
+        signInClient.signOut().addOnCompleteListener(mActivity,
+                new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        boolean successful = task.isSuccessful();
+                        Log.d(ACTIVITY_TAG, "signOut(): " + (successful ? "success" : "failed"));
```

Comment:
Good practice to handle sign-out failures.
```
if (task.isSuccessful()) {
              Log.d(TAG, "signOut(): success");
            } else {
              handleException(task.getException(), "signOut() failed!");
            }
```
See: https://github.com/playgameservices/android-basic-samples/blob/master/CollectAllTheStars2/src/main/java/com/google/example/games/catt2/MainActivity.java